modules/security-understanding-container-validation.adoc

```diff
@@ -0,0 +1,159 @@
+
+. The cluster compares the calculated digest of the image manifest with the known digest.
+If the digest values do not match, the image is rejected without further processing.
+
+. The cluster downloads an image config file and calculates its digest.
+
+. The cluster compares the calculated digest of the image config file with the digest that is provided in the image manifest.
+If the digest values do not match, the image is rejected without further processing.
+
+. For each file, or layer, that comprises the container image, the cluster downloads the file and calculates its digest.
+
+. The cluster compares the calculated digest of each file that makes up the image with the corresponding digest that is provided in the image manifest.
+If the digest values do not match, the image is rejected.
+
+Once the cluster completes all of the previous steps, the integrity of the image has been validated.
+
+[NOTE]
+====
+The image manifest is validated only when the manifest digest is known, which is true in the following cases:
+
+. The image is pulled by specifying the SHA256 image digest, for example `quay.io/openshift-release-dev/ocp-release@sha256:ea7ac3ad42169b39fce07e5e53403a028644810bee9a212e7456074894df40f3`.
+
+. The image signature is also being validated. See “Image authenticity validation” for more information.
+
+However, if the image manifest is not validated in these cases, the image config and layers are still checked for integrity.
+====
+
+[id="security-understanding-image-authenticity_{context}"]
+== Image authenticity validation
+
+Cluster administrators can use `ClusterImagePolicy` resources to declare an image policy for the cluster, requiring signature validation and other policy checks in addition to image integrity.
+
+:FeatureName: Configuring cluster image policies
+include::snippets/technology-preview.adoc[]
+
+.Example `ClusterImagePolicy` resource 1
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1alpha1
+kind: ClusterImagePolicy
+metadata:
+  name: mypolicy-0
+spec:
+  scopes:
+  - test0.com <1>
+  - test1.com
+  policy:
+    rootOfTrust:
+      policyType: PublicKey
+      publicKey:
+        keyData: dGVzdC1rZXktZGF0YQ==
+----
+<1> This policy for `test0.com` and the policy from `mypolicy-1` will be appended together.
+
+
+.Example `ClusterImagePolicy` resource 2
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1alpha1
+kind: ClusterImagePolicy
+metadata:
+  name: mypolicy-1
+spec:
+  scopes:
+  - test0.com
+  policy:
+    rootOfTrust:
+      policyType: FulcioCAWithRekor
+      fulcioCAWithRekor:
+        fulcioCAData: dGVzdC1jYS1kYXRhLWRhdGE=
+        rekorKeyData: dGVzdC1yZWtvci1rZXktZGF0YQ==
+        fulcioSubject: <1>
+          oidcIssuer: https://OIDC.example.com
+          signedEmail: test-user@example.com
+    signedIdentity:
+      matchPolicy: RemapIdentity
+      remapIdentity:
+        prefix: test-remap-prefix
+        signedPrefix: test-remap-signed-prefix
```

Comment:
(For the record, these values are invalid will either be rejected, or won’t do anything — real prefixes contain at least a host name with at least one dot, or a host:port. But then again, the `…Data` fields don’t contain really realistic inputs either.)

Reply:
I think that _if we care_ about the example values, it would be better to make this all realistic. `scopes: … registry.internal`; `prefix: registry.internal/vendorA/productA`; `signedPrefix: vendorA.example/productA` or something (and maybe the same `vendorA.example` in the Fulcio conditions).

But, also, it seems to me that details of these CRs are really not the primary focus of the article.